src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,647 @@
+            // Before:
+            // void Caller() { var x = Callee(); }
+            // long Callee() { return 1 }
+            //
+            // After
+            // void Caller() { var x = (long)1; }
+            // int Callee() { return 1; }
+            //
+            // Case 3 (type cast & additional parenthesis)
+            // // Before:
+            // // void Caller() { var x = Callee()(); }
+            // // Func<int> Callee() { return () => 1; }
+            // // After:
+            // // void Caller() { var x = ((Func<int>)(() => 1))(); }
+            // // Func<int> Callee() { return () => 1; }
+            if (!calleeMethodSymbol.ReturnsVoid && !isThrowExpression)
+            {
+                inlineExpression = (TExpressionSyntax)syntaxGenerator.AddParentheses(
+                    syntaxGenerator.CastExpression(
+                        GenerateTypeSyntax(calleeMethodSymbol.ReturnType, allowVar: false),
+                        syntaxGenerator.AddParentheses(inlineExpression)));
+            }
+
+            return new InlineMethodContext(
+                localDeclarationStatementsNeedInsert,
+                statementContainsCallee,
+                inlineExpression.WithTriviaFrom(calleeInvocationNode),
+                calleeInvocationNode,
+                containsAwaitExpression);
+        }
+
+        private static ImmutableArray<TStatementSyntax> GetLocalDeclarationStatementsNeedInsert(
+            SyntaxGenerator syntaxGenerator,
+            ImmutableArray<(IParameterSymbol parameterSymbol, TExpressionSyntax expression)> parametersToGenerateFreshVariablesFor,
+            ImmutableArray<(IParameterSymbol parameterSymbol, string identifierName)> parametersWithVariableDeclarationArgument,
+            ImmutableDictionary<ISymbol, string> renameTable)
+        {
+            var declarationsQuery = parametersToGenerateFreshVariablesFor
+                .Select(parameterAndArguments => CreateLocalDeclarationStatement(syntaxGenerator, renameTable, parameterAndArguments));
+
+            var declarationsForVariableDeclarationArgumentQuery = parametersWithVariableDeclarationArgument
+                .Select(parameterAndName =>
+                    (TStatementSyntax)syntaxGenerator.LocalDeclarationStatement(
+                        parameterAndName.parameterSymbol.Type,
+                        parameterAndName.identifierName));
+
+            return declarationsQuery.Concat(declarationsForVariableDeclarationArgumentQuery).ToImmutableArray();
+        }
+
+        private bool ContainsAwaitExpression(TExpressionSyntax inlineExpression, TMethodDeclarationSyntax calleeMethodDeclarationNode)
+        {
+            // Check if there is await expression. It is used later if the caller should be changed to async
+            var awaitExpressions = inlineExpression
+                .DescendantNodesAndSelf()
+                .Where(node => node != null && _syntaxFacts.IsAwaitExpression(node))
```

Comment:
```suggestion
                .Where(node => _syntaxFacts.IsAwaitExpression(node))
```

Reply:
note: This is likely descneding too far.  i.e. if you hav `Foo(async() => await Task.Delay())` this should not count as containing an await as you crossed a lambda boundary.

Reply:
Add a predicate to avoid lambda